### ISSUE TITLE
change_a_inc division by zero when there is no change in inclination and a change in the semi-major axis

### ISSUE THREAD

**Issue by jfrchicanog** [poliastro/poliastro#1384]
https://github.com/poliastro/poliastro/blob/5f690dfd9119c16be05c0f364a8d25eb54ab26aa/src/poliastro/core/thrust/change_a_inc.py#L53

For example:

```
>>> from  poliastro.twobody.thrust import change_a_inc
>>> from poliastro.bodies import Sun
>>> from astropy import units as u
>>> a_d, delta_v, t_f = change_a_inc(Sun.k, 150000000*u.km, 200000000*u.km, 12*u.deg, 12*u.deg, 1*u.m/u.s**2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/francis/Documents/investigacion/Proyectos/GTOC 11/gtoc11/poliastro/twobody/thrust/change_a_inc.py", line 42, in change_a_inc
    a_d, delta_V, t_f = change_a_inc_fast(
  File "/Users/francis/Documents/investigacion/Proyectos/GTOC 11/gtoc11/poliastro/core/thrust/change_a_inc.py", line 109, in change_a_inc
    delta_V, t_f = extra_quantities(k, a_0, a_f, inc_0, inc_f, f)
ZeroDivisionError: division by zero

```

**Comment by astrojuanlu** [comment]
I can reproduce. Temporarily disabling the Numba JIT and making the NumPy errors more strict, we get some more information:

```
$ NUMBA_DISABLE_JIT=1 ipython --no-banner

In [1]: import numpy as np

In [2]: np.seterr(invalid="raise")
Out[2]: {'divide': 'warn', 'over': 'warn', 'under': 'ignore', 'invalid': 'warn'}

In [3]: from poliastro.twobody.thrust import change_a_inc
   ...: from poliastro.bodies import Sun
   ...: from astropy import units as u
   ...: a_d, delta_v, t_f = change_a_inc(Sun.k, 150000000*u.km, 200000000*u.km, 12*u.deg, 12*u.deg, 1*u.m/u.s**2)
---------------------------------------------------------------------------
FloatingPointError                        Traceback (most recent call last)
<ipython-input-3-f2eb190e8e89> in <module>
      2 from poliastro.bodies import Sun
      3 from astropy import units as u
----> 4 a_d, delta_v, t_f = change_a_inc(Sun.k, 150000000*u.km, 200000000*u.km, 12*u.deg, 12*u.deg, 1*u.m/u.s**2)

~/Projects/LSF/poliastro/library/src/poliastro/twobody/thrust/change_a_inc.py in change_a_inc(k, a_0, a_f, inc_0, inc_f, f)
     40       Transfer Problem Using Optimal Control Theory", 1997.
     41     """
---> 42     a_d, delta_V, t_f = change_a_inc_fast(
     43         k=k.to_value(u.km ** 3 / u.s ** 2),
     44         a_0=a_0.to_value(u.km),

~/Projects/LSF/poliastro/library/src/poliastro/core/thrust/change_a_inc.py in change_a_inc(k, a_0, a_f, inc_0, inc_f, f)
    107         return accel_v
    108 
--> 109     delta_V, t_f = extra_quantities(k, a_0, a_f, inc_0, inc_f, f)
    110     return a_d, delta_V, t_f

~/Projects/LSF/poliastro/library/src/poliastro/core/thrust/change_a_inc.py in extra_quantities(k, a_0, a_f, inc_0, inc_f, f)
     11     """Extra quantities given by the Edelbaum (a, i) model."""
     12     V_0, beta_0_ = compute_parameters(k, a_0, a_f, inc_0, inc_f)
---> 13     delta_V_ = delta_V(V_0, beta_0_, inc_0, inc_f)
     14     t_f_ = delta_V_ / f
     15 

~/Projects/LSF/poliastro/library/src/poliastro/core/thrust/change_a_inc.py in delta_V(V_0, beta_0, inc_0, inc_f)
     46     """Compute required increment of velocity."""
     47     delta_i_f = abs(inc_f - inc_0)
---> 48     return V_0 * np.cos(beta_0) - V_0 * np.sin(beta_0) / np.tan(
     49         np.pi / 2 * delta_i_f + beta_0
     50     )

FloatingPointError: invalid value encountered in double_scalars

In [4]: %debug
> /home/juanlu/Projects/LSF/poliastro/library/src/poliastro/core/thrust/change_a_inc.py(48)delta_V()
     46     """Compute required increment of velocity."""
     47     delta_i_f = abs(inc_f - inc_0)
---> 48     return V_0 * np.cos(beta_0) - V_0 * np.sin(beta_0) / np.tan(
     49         np.pi / 2 * delta_i_f + beta_0
     50     )

ipdb> p delta_i_f
0.0
ipdb> p beta_0
0.0
ipdb> p np.tan(np.pi / 2 * delta_i_f + beta_0)
0.0
```

We should make the code more robust against this case.

**Comment by astrojuanlu** [comment]
When `delta_i_f = 0`, this expression

https://github.com/poliastro/poliastro/blob/764a6b66a87d811d576e35a8ec497d270e54c128/src/poliastro/core/thrust/change_a_inc.py#L48-L50

(which, by the way, is Eq. 43 of Kéchichian, 1997)

![Screenshot 2021-11-05 at 15-50-20 Reformulation of Edelbaum's Low-Thrust Transfer Problem Using Optimal Control Theory - ke](https://user-images.githubusercontent.com/316517/140530030-f827d3c9-e1a3-4110-93dd-ae3b34fb16a1.png)

simplifies to `V_0 * np.cos(beta_0) - V_0 * np.cos(beta_0) = 0`. This doesn't make sense, because as I wrote in [my final Masters project](https://github.com/astrojuanlu/pfc-uc3m/blob/master/study-analytical-solutions.pdf):

![Screenshot 2021-11-05 at 15-43-05 study-analytical-solutions pdf](https://user-images.githubusercontent.com/316517/140528913-7fd0301e-382a-4fe3-affa-0dbc06413997.png)

and as mentioned also at the end of section III of Kéchichian, 1997:

![Screenshot 2021-11-05 at 16-00-45 Reformulation of Edelbaum's Low-Thrust Transfer Problem Using Optimal Control Theory - ke](https://user-images.githubusercontent.com/316517/140531779-7ab8aadf-77e1-4fee-8315-4ca0a10e800a.png)

Maybe there's a very obvious mathematical detail I'm not seeing, but I don't understand how the formulas gracefully degrade to the planar case when the change of inclination is zero.

**Comment by astrojuanlu** [comment]
From section IV of Kéchichian, 1997:

![Screenshot 2021-11-05 at 16-10-42 Reformulation of Edelbaum's Low-Thrust Transfer Problem Using Optimal Control Theory - ke](https://user-images.githubusercontent.com/316517/140533320-17405e27-6cb7-4228-9e64-066ebfe0285e.png)

Therefore: yes, the assumptions of the model don't hold when $\Delta i = 0$ ✔️

For now we could include a better error message catching this condition early on, but the interesting question is how to retrieve the guidance law in this case.

**Comment by jfrchicanog** [comment]
I think the case when Delta i = 0 is easy. In that case beta=0 for all t and the time required is Delta V / t, where Delta V is the absolute value of the difference in the velocities for the different semimajor axes.

Francis

**Comment by MLopez-Ibanez** [comment]
This paper presents two alternative ways to change only a: https://link.springer.com/article/10.1007/s10569-021-10033-9#Sec3

**Comment by astrojuanlu** [comment]
Fixed in #1390

**Comment by astrojuanlu** [comment]
> This paper presents two alternative ways to change only a: https://link.springer.com/article/10.1007/s10569-021-10033-9#Sec3

That looks familiar 😃 https://mobile.twitter.com/poliastro_py/status/1416819312157990920 

I had forgotten about it. Opening a new issue to review it and implement it.
